tools/code_format/check_format.py

```diff
@@ -765,44 +861,38 @@ def fix_build_line(self, file_path, line, line_number):
 
     def fix_build_path(self, file_path):
         self.evaluate_lines(file_path, functools.partial(self.fix_build_line, file_path))
-
         error_messages = []
 
         # TODO(htuch): Add API specific BUILD fixer script.
-        if not self.is_build_fixer_excluded_file(file_path) and not self.is_api_file(
-                file_path) and not self.is_starlark_file(file_path) and not self.is_workspace_file(
-                    file_path):
-            if os.system("%s %s %s" %
-                         (self.config.paths["build_fixer_py"], file_path, file_path)) != 0:
-                error_messages += ["envoy_build_fixer rewrite failed for file: %s" % file_path]
-
-        if os.system("%s -lint=fix -mode=fix %s" % (self.config.buildifier_path, file_path)) != 0:
-            error_messages += ["buildifier rewrite failed for file: %s" % file_path]
+        if self._run_build_fixer(file_path):
+            fixer_command = f"{self.config.paths['build_fixer_py']} {file_path} {file_path}"
+            if os.system(fixer_command) != 0:
+                error_messages.append(f"envoy_build_fixer rewrite failed for file: {file_path}")
+
+        buildifier_command = f"{self.config.buildifier_path} -lint=fix -mode=fix {file_path}"
+        if os.system(buildifier_command) != 0:
+            error_messages.append(f"buildifier rewrite failed for file: {file_path}")
         return error_messages
 
     def check_build_path(self, file_path):
         error_messages = []
-
-        if not self.is_build_fixer_excluded_file(file_path) and not self.is_api_file(
-                file_path) and not self.is_starlark_file(file_path) and not self.is_workspace_file(
-                    file_path):
-            command = "%s %s | diff %s -" % (
-                self.config.paths["build_fixer_py"], file_path, file_path)
-            error_messages += self.execute_command(
-                command, "envoy_build_fixer check failed", file_path)
-
-        if self.is_build_file(file_path) and file_path.startswith(self.api_prefix + "envoy"):
+        if self._run_build_fixer(file_path):
+            command = f"{self.config.paths['build_fixer_py']} {file_path} | diff {file_path} -"
+            error_messages.extend(
+                self.execute_command(command, "envoy_build_fixer check failed", file_path))
+        envoy_api_build_file = (
+            self.is_build_file(file_path) and file_path.startswith(f"{self.api_prefix}envoy"))
```

Comment:
nit: consider failing on args processing if `api_prefix` doesn't end with `/`, or normalizing it.

Reply:
this was the original logic - im reluctant to make any changes to that right now but its totally on my priority list to improve the logic here

Reply:
I missed that this was originally here. Agreed.